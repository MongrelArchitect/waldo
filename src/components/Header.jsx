import React from 'react';
import Nav from './Nav';
import Message from './Message';
import Timer from './Timer';
import Characters from './Characters';

export default function Header({
  characters, found, messageVisible, time,
}) {
  return (
    <header>
      <Nav />
      <Message found={found} visible={messageVisible} />
      <Timer time={time} />
      <Characters characters={characters} />
    </header>
  );
}
