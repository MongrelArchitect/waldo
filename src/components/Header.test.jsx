import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Header from './Header';

describe('Header component', () => {
  const chars = [
    { name: 'char one', img: '#', found: false },
    { name: 'char two', img: '##', found: false },
    { name: 'char three', img: '###', found: true },
  ];

  it('exists', () => {
    render(
      <BrowserRouter>
        <Header characters={chars} />
      </BrowserRouter>,
    );
    expect(screen.getByRole('banner')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Header characters={chars} />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
