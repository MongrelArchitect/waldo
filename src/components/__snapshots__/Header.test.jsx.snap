// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div>
  <header>
    <nav>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        class=""
        href="/play"
      >
        Play
      </a>
      <a
        class=""
        href="/scores"
      >
        Scoreboard
      </a>
    </nav>
    <span
      class="timer"
    >
      NaN:NaN.NaN
    </span>
    <span
      class="characters"
      data-testid="characters"
    >
      <img
        alt="char one"
        class="not-found"
        src="#"
      />
      <img
        alt="char two"
        class="not-found"
        src="##"
      />
      <img
        alt="char three"
        class="found"
        src="###"
      />
    </span>
  </header>
</div>
`;
